### PR TITLE
#MLOPS- 80 Add model path formatting, send iteration to api functionality

### PR DESCRIPTION
**Pull request by kajetsz**

### CODE REVIEW

File: library/mlops/src/iteration.py [hunk 1]
```diff
@@ -33,7 +33,10 @@ def log_path_to_model(self, path_to_model: str):
         """
         # TODO: we need to take as an input path with single backslashes (e.g. "C:\mlops-inzynierka\LICENSE") and
         # use some regex to parse it to double backslashes, because JSON doesn't accept single ones
-        raise NotImplementedError
+        # add universal path support :)
+        #raise NotImplementedError
+
+        self.path_to_model = path_to_model.replace('/', '\\').replace('\f', '\\f').replace('\t', '\\t').replace('\n', '\\n').replace('\r', '\\r').replace('\b', '\\b').replace('\\','\\\\')
```
Comment (directtt) [inline library/mlops/src/iteration.py:39]: it's better to separate this to different function

File: library/mlops/tracking.py [hunk 2]
```diff
@@ -168,6 +168,12 @@ def start_iteration(iteration_name: str, project_id: str = None, experiment_id:
     # TODO: add exceptions when one of id = None or id does not exists
     # In general, we can create separate folder exceptions, same as it's done in back-end routers,
     # because we need to also add some exceptions for further functions
+
+    if not project_id:
```
Comment (directtt) [inline library/mlops/tracking.py:172]: additionaly, we can use these expections in previous functions, which use `active_project_id` or `active_experiment_id` to check if it's not None, for example in `create_experiment()` function may occur an exception

**Review by directtt — CHANGES_REQUESTED** [review]
looks good, some minor change, I think it's better to leave a review than chaning it by myself

also, pls change branch name to `feature/` and PR title to stay in previous conention

File: library/mlops/tracking.py (hunk 2 above)
Comment (directtt) [reply library/mlops/tracking.py:172]: Let's also change outputs of `set_active_project` and `set_active_experiment` from raw dictionary to some pretty-print string like : "Active project set to: {project_id}"

File: library/mlops/exceptions/tracking.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+def project_id_missing_exception():
```
Comment (Kacperxxx) [inline library/mlops/exceptions/tracking.py:2]: To bo honest, project_id_missing_exception() is ambiguous. Firstly, project_id is None and secondly project with this id does not exist. In my opinion this should be split into two exceptions: project_id_not_specified and project_id_not_found.

File: library/mlops/exceptions/tracking.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    return ValueError('Project ID is missing')
+
```
Comment (Kacperxxx) [inline library/mlops/exceptions/tracking.py:4]: To bo honest, project_id_missing_exception() is ambiguous. Firstly, project_id is None and secondly project with this id does not exist. In my opinion this should be split into two exceptions: project_id_not_specified and project_id_not_found.

File: library/mlops/exceptions/tracking.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+def experiment_id_missing_exception():
+    return ValueError('Experiment ID is missing')
```
Comment (Kacperxxx) [inline library/mlops/exceptions/tracking.py:6]: To bo honest, experiment_id_missing_exception() is ambiguous. Firstly, experiment_id is None and secondly experiment with this id does not exist. In my opinion this should be split into two exceptions: experiment_id_not_specified and experiment_id_not_found.

**Review by Kacperxxx — CHANGES_REQUESTED** [review]
There are ambiguous exceptions here that I believe need to be corrected.

**Pull request closed without merging**